userspace/libscap/scap_open.h

```diff
@@ -59,11 +59,13 @@ typedef enum {
 	 * Do not read system call data. Events come from the configured input plugin.
 	 */
 	SCAP_MODE_PLUGIN,
+#ifdef HAS_ENGINE_MODERN_BPF
```

Comment:
@Andreagit97 I had to ifdef this out to be consistent with the feature gate. However, I think putting the modern probe in this enum sort of breaks the pattern we had so far in libscap. I think the new probe could be under the `SCAP_MODE_LIVE` umbrella and be configurable from the `scap_open_args`.

My call would be to keep things as they are for now, and eventually think about this when the modern probe will be full-functioning, WDYT?

Reply:
First of all, thank you for the fix, I missed this feature gate! Completely agree with you, this was just a workaround to build things since the `v-table` was not complete at that time. My final idea is to have an enum inside the `scap-open` args that allow you to choose which driver to use and set again the new probe under `SCAP_MODE_LIVE` as you said. 

Agree also on the second point, when we will have something full-working we will adjust all these details to respect the actual conventions :) 

Thank you very much @jasondellaluce!

Reply:
I also agree with `SCAP_MODE_LIVE`, as that's the mode that kmod and current bpf use!